Test/QuickCheck/Modifiers.hs

```diff
@@ -343,4 +355,100 @@ instance (Arbitrary a, ShrinkState s a) => Arbitrary (Shrinking s a) where
 #endif /* NO_MULTI_PARAM_TYPE_CLASSES */
 
 --------------------------------------------------------------------------
+-- | @ASCII@.
+newtype ASCII = ASCII {getASCII :: Char}
+  deriving ( Eq, Ord, Show, Read )
+
+asciiGen :: Gen Char
+asciiGen = choose ('\0', '\127')
+
+instance Arbitrary ASCII where
+  arbitrary = ASCII `fmap` asciiGen
+
+--------------------------------------------------------------------------
+-- | @Latin1@: latin1 Char, same Arbitrary as Char itself.
+newtype Latin1 = Latin1 {getLatin1 :: Char}
```

Comment:
For me it's surprising that `Char` instance generates only `Latin1` characters. Could you add a haddock to `Arbitrary Char` saying it?

Also how the `-----` top bar affects the haddocks?